File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
     let ans=[];
        let neg=1;
        let pos=0;
        for(let i =0 ; i<nums.length;i++){
            if (nums[i] > 0){
                ans[pos]=nums[i];
                pos+=2;
            }else{
                ans[neg]=nums[i];
                neg+=2;
            }
        }
            
        return ans;
};

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        ans=[0]* len(nums)
        neg=1
        pos=0
        for i in nums:
            if i > 0:
                ans[pos]=i
                pos+=2
            else:
                ans[neg]=i
                neg+=2
        return ans 